butane_codegen/src/lib.rs

```diff
@@ -167,3 +167,39 @@ fn migrations_dir() -> PathBuf {
     dir.push("migrations");
     dir
 }
+
+#[proc_macro_derive(ButaneJson)]
```

Comment:
If this approach is the right one for now, the name `ButaneJson` needs to change as the `Butane..` is redundant. The new naming should allow for another derive macros, such as for enums https://github.com/Electron100/butane/issues/20

One alternative is to have something like `#[butane_type(Json, derive=true)` automatically do this, and `#[butane_type(Text, derive=true)]` would be the equivalent for enums.  This at least avoids a bunch of new names.  But is `butane_type` going to survive after https://github.com/Electron100/butane/commits/migration_v2 ?

Happy to try re-implementing this any other way that is suggested.

Reply:
I think conceptually what we're doing here is deriving `FieldType` (and its dependencies `ToSql`/`FromSql`) to enable a type to be used in model fields. However, it's conceptually a parameterized derive (Json is the parameter) which Rust doesn't really have a concept of today. So I might be inclined to define this as an attribute macro (because it's more flexible than a derive macro) something like `#[field_type(Json)]`, or maybe `#DeriveFieldType(Json)`.

It's arguably clearer to be able to write `#[Derive(..)]` here, but since `ButaneJson` isn't a real trait, I'm not sure `#[Derive(ButaneJson)]` would really help clarity

I'm not one to get super hung up on names though.

`butane_type` is not expected to survive migration v2. I've filed #60 to track that issue/experiment more clearly

Reply:
I have updated this to be a derive of `FieldType` - it currently only handles `butane_type` of `Json` but it could now be expanded to handle other types.